core/state/statedb_test.go

```diff
@@ -197,16 +197,24 @@ func TestCopy(t *testing.T) {
 	}
 
 	// Finalise the changes on all concurrently
-	finalise := func(wg *sync.WaitGroup, db *StateDB) {
-		defer wg.Done()
+	finalise := func(db *StateDB) {
 		db.Finalise(true)
 	}
 
 	var wg sync.WaitGroup
 	wg.Add(3)
-	go finalise(&wg, orig)
-	go finalise(&wg, copy)
-	go finalise(&wg, ccopy)
+	go func() {
+		defer wg.Done()
+		finalise(orig)
```

Comment:
The original was written using a closure, `finalise`, to be concise. You've undone that, so you might aswell can drop the `finalize` closure, and change this to 
```suggestion
		orig.Finalize(true)
```
And further below `copy.Finalize(true)` and `ccopy.Finalise(true)`.

Reply:
Done~